x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/PushExpressionsToFieldLoad.java

```diff
@@ -113,12 +113,6 @@ private static Expression replaceFieldsForFieldTransformations(
             new NameId(),
             true
         );
-        Attribute.IdIgnoringWrapper key = newFunctionAttr.ignoreId();
-        if (addedAttrs.containsKey(key)) {
-            return addedAttrs.get(key);
-        }
-
-        addedAttrs.put(key, newFunctionAttr);
-        return newFunctionAttr;
+        return addedAttrs.computeIfAbsent(newFunctionAttr.ignoreId(), k -> newFunctionAttr);
```

Comment:
This is unrelated too. I got to it as I was considering adding an IdIgnoringAttributeSet and was inspecting other places where this would help, then noticed this can be made easier to read.